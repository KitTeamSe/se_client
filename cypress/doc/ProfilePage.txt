프로필 화면 관련 E2E 테스트
  - 로그인 후 프로필 화면 이동
  - 10개 정보가 있는지 확인
  - 개인정보 수정모드시 3개의 input 이 있는지 확인
  - 취소버튼 테스트
  - 닉네임에 아무거나 입력 후 새로고침버튼 클릭 후 돌아오는지 확인
  - 정보공유동의를 수정
  - 수정되었는지 확인
  - 비밀번호 변경 테스트