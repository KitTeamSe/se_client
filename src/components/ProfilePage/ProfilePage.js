import React from 'react';
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
  Button,
  Menu,
  Select,
  MenuItem,
  TextField,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText
} from '@material-ui/core';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faTools,
  faTimesCircle,
  faSyncAlt
} from '@fortawesome/free-solid-svg-icons';

import {
  accountData,
  changebleAccount,
  informationOpenAgreeEnum,
  typeList
} from '../../DataExport';

const Wrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;

const InfoTableWrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  width: 100%;
  max-width: 396px;
`;

const InfoTable = styled(TableContainer)`
  margin-top: 24px;
`;

const Welcome = styled.div`
  margin: 24px;
  font-size: 24px;
`;

const MyinfoHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  position: relative;
`;

const EditTableCell = styled.th`
  text-align: right;
  padding: 6px;
  display: table-cell;
  font-size: 0.875rem;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 400;
  line-height: 1.43;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  letter-spacing: 0.01071em;
  vertical-align: inherit;
`;

const FormInput = styled.input`
  size: 20;
  height: 15px;
  border: none;
  padding: 12px;
  text-align: right;
  font-size: 0.875rem;
  font-weight: bold;
  autocomplete: off;
  &:focus {
    outline: none;
    border: none;
  }
`;

const FormField = styled.form`
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  align-items: center;
`;

const FormTextField = styled(TextField)`
  margin: 12px 32px 12px 32px;
  min-width: 256px;
`;

const ErrorText = styled.div`
  margin: 6px;
  font-size: 18px;
  color: red;
`;

const FormSelectField = styled(Select)`
  margin-right: 2px;
  width: 128px;
`;

const ButtonWrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
`;

const HalfButton = styled(Button)`
  width: 50%;
  padding: 8px;
  margin: 8px;
`;

const RefreshIcon = styled(FontAwesomeIcon)`
  position: absolute;
  right: -10px;
  transition: 0.5s;
  &:hover {
    transform: rotate(180deg);
    transition: 1s;
  }
`;

const PwChangeDialog = props => {
  const { mode, error, newPwForm, modeChange, pwChangeSubmit, formChange } =
    props;
  return (
    <>
      <Dialog
        open={mode === 'pwChangeMode'}
        onClose={modeChange}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">비밀번호 변경</DialogTitle>
        <FormField onSubmit={pwChangeSubmit}>
          <FormTextField
            autoFocus
            id="nowPassword"
            label="현재 비밀번호를 입력하세요"
            type="password"
            onChange={formChange}
            value={newPwForm.nowPassword}
          />
          <FormTextField
            id="newPassword"
            label="새로운 비밀번호"
            type="password"
            onChange={formChange}
            value={newPwForm.newPassword}
          />
          <FormTextField
            id="newPasswordConfirm"
            label="새로운 비밀번호 확인"
            type="password"
            onChange={formChange}
            value={newPwForm.newPasswordConfirm}
          />
          <ErrorText>{error}</ErrorText>
          <ButtonWrapper>
            <HalfButton onClick={modeChange} color="primary">
              취소
            </HalfButton>
            <HalfButton type="submit" onClick={pwChangeSubmit} color="primary">
              변경하기
            </HalfButton>
          </ButtonWrapper>
        </FormField>
      </Dialog>
    </>
  );
};

const WithdrawalDialog = props => {
  const {
    mode,
    error,
    withDrawalForm,
    modeChange,
    formChange,
    withdrawalSubmit
  } = props;
  return (
    <>
      <Dialog
        open={mode === 'withdrawalMode'}
        onClose={modeChange}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">회원 탈퇴</DialogTitle>
        <DialogContent>
          <DialogContentText> 나 탈퇴한다~~~~~~~~~</DialogContentText>
        </DialogContent>
        <FormField submit={withdrawalSubmit}>
          <FormTextField
            autoFocus
            id="password"
            label="비밀번호를 입력하세요"
            type="password"
            onChange={formChange}
            value={withDrawalForm.password}
          />
          <FormTextField
            id="text"
            label="'탈퇴'를 입력하세요"
            type="text"
            onChange={formChange}
            value={withDrawalForm.text}
          />
          <Wrapper>
            <ErrorText style={{ fontSize: '16px' }}>{error}</ErrorText>
          </Wrapper>
          <ButtonWrapper>
            <HalfButton onClick={modeChange} color="primary">
              취소
            </HalfButton>
            <HalfButton
              type="submit"
              onClick={withdrawalSubmit}
              color="primary"
            >
              탈퇴
            </HalfButton>
          </ButtonWrapper>
        </FormField>
      </Dialog>
    </>
  );
};

const ProfileHeader = props => {
  const {
    mode,
    anchorEl,
    error,
    newPwForm,
    withDrawalForm,
    modeChange,
    menuClick,
    formChange,
    pwChangeSubmit,
    withdrawalSubmit,
    editFormRefresh
  } = props;

  return (
    <MyinfoHeader>
      {mode === 'editMode' ? (
        <>
          <div />
          <Welcome>기본정보</Welcome>
          <FontAwesomeIcon
            icon={faTimesCircle}
            size="lg"
            color="#DC143C"
            style={{ cursor: 'pointer' }}
            onClick={modeChange}
          />
          <RefreshIcon
            icon={faSyncAlt}
            size="lg"
            style={{ cursor: 'pointer' }}
            onClick={editFormRefresh}
          />
        </>
      ) : (
        <>
          <div />
          <Welcome>기본정보</Welcome>
          <FontAwesomeIcon
            icon={faTools}
            size="lg"
            style={{ cursor: 'pointer' }}
            onClick={menuClick}
          />
          <Menu
            id="editMode"
            anchorEl={anchorEl}
            keepMounted
            open={Boolean(anchorEl)}
            onClose={menuClick}
            style={{ marginLeft: '36px' }}
          >
            <MenuItem id="editMode" onClick={modeChange}>
              개인정보 수정
            </MenuItem>
            <MenuItem id="pwChangeMode" onClick={modeChange}>
              비밀번호 변경
            </MenuItem>
            <MenuItem id="withdrawalMode" onClick={modeChange}>
              회원탈퇴
            </MenuItem>
          </Menu>
        </>
      )}
      <PwChangeDialog
        mode={mode}
        error={error}
        newPwForm={newPwForm}
        modeChange={modeChange}
        formChange={formChange}
        pwChangeSubmit={pwChangeSubmit}
      />
      <WithdrawalDialog
        mode={mode}
        withDrawalForm={withDrawalForm}
        error={error}
        modeChange={modeChange}
        formChange={formChange}
        withdrawalSubmit={withdrawalSubmit}
      />
    </MyinfoHeader>
  );
};

const ProfileRow = props => {
  const { row } = props;
  return (
    <>
      <TableRow key={row[0]}>
        <TableCell component="th" scope="row">
          {accountData[row[0]]}
        </TableCell>
        {row[0] === 'informationOpenAgree' ? (
          <TableCell align="right">
            {informationOpenAgreeEnum[row[1]]}
          </TableCell>
        ) : (
          <TableCell align="right">{row[1]}</TableCell>
        )}
      </TableRow>
    </>
  );
};

const EditRowClassifier = props => {
  const {
    row,
    informationOpenAgreeChange,
    infoEditObj,
    handleChange,
    typeChange
  } = props;
  const editRowValue = infoEditObj[row[0]];
  if (row[0] === 'informationOpenAgree') {
    return (
      <EditTableCell
        align="right"
        onClick={informationOpenAgreeChange}
        style={{
          paddingRight: '16px',
          cursor: 'pointer'
        }}
      >
        {informationOpenAgreeEnum[editRowValue]}
      </EditTableCell>
    );
  }
  if (row[0] === 'type') {
    return (
      <EditTableCell>
        <FormSelectField
          name="editType"
          id={row[0]}
          value={editRowValue}
          onChange={typeChange}
        >
          {typeList.map(type => (
            <MenuItem value={type.userType} key={type.typeid}>
              {type.userType}
            </MenuItem>
          ))}
        </FormSelectField>
      </EditTableCell>
    );
  }
  return (
    <EditTableCell>
      <FormInput
        name={accountData[row[0]]}
        id={row[0]}
        value={editRowValue}
        onChange={handleChange}
      />
    </EditTableCell>
  );
};

const EditRow = props => {
  const {
    row,
    infoEditObj,
    informationOpenAgreeChange,
    handleChange,
    typeChange
  } = props;
  return (
    <TableRow key={row[0]}>
      <TableCell component="th" scope="row">
        {accountData[row[0]]}
      </TableCell>
      <EditRowClassifier
        row={row}
        infoEditObj={infoEditObj}
        informationOpenAgreeChange={informationOpenAgreeChange}
        handleChange={handleChange}
        typeChange={typeChange}
      />
    </TableRow>
  );
};

const SubmitButton = props => {
  const { mode, myinfoEditSubmit } = props;
  return (
    <>
      {mode === 'editMode' ? (
        <Button
          variant="contained"
          color="primary"
          type="submit"
          style={{ margin: '24px', cursor: 'pointer' }}
          onClick={myinfoEditSubmit}
        >
          수정
        </Button>
      ) : (
        <></>
      )}
    </>
  );
};

const PropfilePage = props => {
  const {
    infoObj,
    infoEditObj,
    anchorEl,
    newPwForm,
    error,
    withDrawalForm,
    mode
  } = props;

  const {
    handleChange,
    modeChange,
    informationOpenAgreeChange,
    myinfoEditSubmit,
    menuClick,
    formChange,
    pwChangeSubmit,
    typeChange,
    withdrawalSubmit,
    editFormRefresh
  } = props;

  const rows = Object.entries(infoObj);

  return (
    <Wrapper>
      <InfoTableWrapper>
        <ProfileHeader
          mode={mode}
          error={error}
          anchorEl={anchorEl}
          newPwForm={newPwForm}
          withDrawalForm={withDrawalForm}
          formChange={formChange}
          modeChange={modeChange}
          menuClick={menuClick}
          pwChangeSubmit={pwChangeSubmit}
          withdrawalSubmit={withdrawalSubmit}
          editFormRefresh={editFormRefresh}
        />
        <InfoTable>
          <Table>
            <TableBody>
              {rows.map(row => (
                <>
                  {changebleAccount.includes(row[0]) && mode === 'editMode' ? (
                    <EditRow
                      row={row}
                      infoEditObj={infoEditObj}
                      informationOpenAgreeChange={informationOpenAgreeChange}
                      handleChange={handleChange}
                      typeChange={typeChange}
                    />
                  ) : (
                    <ProfileRow row={row} />
                  )}
                </>
              ))}
            </TableBody>
          </Table>
        </InfoTable>
        <ErrorText>{error}</ErrorText>
        <SubmitButton mode={mode} myinfoEditSubmit={myinfoEditSubmit} />
      </InfoTableWrapper>
    </Wrapper>
  );
};

export default PropfilePage;
