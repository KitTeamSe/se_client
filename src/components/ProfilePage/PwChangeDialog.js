import React from 'react';
import { Dialog, DialogTitle, TextField, Button } from '@material-ui/core';
import styled from 'styled-components';

const FormField = styled.form`
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  align-items: center;
`;

const FormTextField = styled(TextField)`
  margin: 12px 32px 12px 32px;
  min-width: 256px;
`;

const ErrorText = styled.div`
  margin: 6px;
  font-size: 18px;
  color: red;
`;

const ButtonWrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
`;

const HalfButton = styled(Button)`
  width: 50%;
  padding: 8px;
  margin: 8px;
`;

const PwChangeDialog = props => {
  const { mode, error, newPwForm, modeChange, pwChangeSubmit, formChange } =
    props;
  return (
    <>
      <Dialog
        open={mode === 'pwChangeMode'}
        onClose={modeChange}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">비밀번호 변경</DialogTitle>
        <FormField onSubmit={pwChangeSubmit}>
          <FormTextField
            autoFocus
            id="nowPassword"
            label="현재 비밀번호를 입력하세요"
            type="password"
            onChange={formChange}
            value={newPwForm.nowPassword}
          />
          <FormTextField
            id="newPassword"
            label="새로운 비밀번호"
            type="password"
            onChange={formChange}
            value={newPwForm.newPassword}
          />
          <FormTextField
            id="newPasswordConfirm"
            label="새로운 비밀번호 확인"
            type="password"
            onChange={formChange}
            value={newPwForm.newPasswordConfirm}
          />
          <ErrorText>{error}</ErrorText>
          <ButtonWrapper>
            <HalfButton onClick={modeChange} color="primary">
              취소
            </HalfButton>
            <HalfButton type="submit" onClick={pwChangeSubmit} color="primary">
              변경하기
            </HalfButton>
          </ButtonWrapper>
        </FormField>
      </Dialog>
    </>
  );
};

export default PwChangeDialog;
