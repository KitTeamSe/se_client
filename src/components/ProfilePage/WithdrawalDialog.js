import React from 'react';
import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  TextField,
  Button
} from '@material-ui/core';
import styled from 'styled-components';

const FormField = styled.form`
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  align-items: center;
`;

const FormTextField = styled(TextField)`
  margin: 12px 32px 12px 32px;
  min-width: 256px;
`;

const ButtonWrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
`;

const HalfButton = styled(Button)`
  width: 50%;
  padding: 8px;
  margin: 8px;
`;

const WithdrawalDialog = props => {
  const { mode, withDrawalForm, modeChange, formChange, withdrawalSubmit } =
    props;
  return (
    <>
      <Dialog
        open={mode === 'withdrawalMode'}
        onClose={modeChange}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">회원 탈퇴</DialogTitle>
        <DialogContent>
          <DialogContentText> 나 탈퇴한다~~~~~~~~~</DialogContentText>
        </DialogContent>
        <FormField submit={withdrawalSubmit}>
          <FormTextField
            autoFocus
            id="password"
            label="비밀번호를 입력하세요"
            type="password"
            onChange={formChange}
            value={withDrawalForm.password}
          />
          <FormTextField
            id="text"
            label="'탈퇴'를 입력하세요"
            type="text"
            onChange={formChange}
            value={withDrawalForm.text}
          />
          <ButtonWrapper>
            <HalfButton onClick={modeChange} color="primary">
              취소
            </HalfButton>
            <HalfButton
              type="submit"
              onClick={withdrawalSubmit}
              color="primary"
            >
              탈퇴
            </HalfButton>
          </ButtonWrapper>
        </FormField>
      </Dialog>
    </>
  );
};

export default WithdrawalDialog;
