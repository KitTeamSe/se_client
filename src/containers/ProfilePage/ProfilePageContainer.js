import React, { useState, useEffect } from 'react';
import { withRouter } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import ProfilePage from '../../components/ProfilePage/ProfilePage';
import {
  myinfo,
  myinfoedit,
  initializeForm,
  changeField,
  accountdelete
} from '../../modules/account';
import WithdrawalDialog from '../../components/ProfilePage/WithdrawalDialog';
import PwChangeDialog from '../../components/ProfilePage/PwChangeDialog';

const ProfilePageContainer = props => {
  const { match, history } = props;

  const [infoObj, setInfoObj] = useState(null);
  const [infoEditObj, setInfoEditObj] = useState({});
  const [anchorEl, setAnchorEl] = useState(null);
  const [mode, setMode] = useState(null);
  const [error, setError] = useState(null);

  const dispatch = useDispatch();
  const token = localStorage.getItem('token');
  const userId = localStorage.getItem('userId');

  if (token === null) {
    history.push('/');
  }

  const {
    data,
    loading,
    myinfoError,
    myinfoEditRes,
    myinfoEditError,
    newPwForm,
    withDrawalForm
  } = useSelector(({ account }) => ({
    data: account.myinfo.data,
    loading: account.myinfo.loading,
    myinfoError: account.myinfo.error,
    myinfoEditRes: account.myinfoEditRes.data,
    myinfoEditError: account.myinfoEditRes.error,
    newPwForm: account.newPwForm,
    withDrawalForm: account.withDrawalForm
  }));

  useEffect(() => {
    const id = match.params.userId;
    dispatch(myinfo({ id }));
  }, [match.params]);

  const editFormRefresh = () => {
    if (data) {
      const res = data.data;
      setInfoObj(res);
      setInfoEditObj(res);
    }
  };

  useEffect(() => {
    editFormRefresh();
  }, [data]);

  useEffect(() => {
    if (myinfoError) {
      setError(String(myinfoError));
    }
    if (myinfoEditError) {
      setError(String(myinfoEditError));
    }
  }, [myinfoError, myinfoEditError]);

  useEffect(() => {
    if (myinfoEditRes) {
      const id = match.params.userId;
      dispatch(myinfo({ id }));
      setMode(null);
    }
  }, [myinfoEditRes]);

  const handleChange = e => {
    e.preventDefault();
    const { value, id } = e.target;
    setInfoEditObj({ ...infoEditObj, [id]: value });
  };

  const typeChange = e => {
    e.preventDefault();
    const { value } = e.target;
    setInfoEditObj({ ...infoEditObj, type: value });
  };

  const informationOpenAgreeChange = e => {
    e.preventDefault();
    if (infoEditObj.informationOpenAgree === 'AGREE') {
      setInfoEditObj({
        ...infoEditObj,
        informationOpenAgree: 'DISAGREE'
      });
    } else {
      setInfoEditObj({
        ...infoEditObj,
        informationOpenAgree: 'AGREE'
      });
    }
  };

  const formChange = e => {
    e.preventDefault();
    const { value, id } = e.target;
    if (['nowPassword', 'newPassword', 'newPasswordConfirm'].includes(id)) {
      dispatch(
        changeField({
          form: 'newPwForm',
          key: id,
          value
        })
      );
    }
    if (['password', 'text'].includes(id)) {
      dispatch(
        changeField({
          form: 'withDrawalForm',
          key: id,
          value
        })
      );
    }
  };

  const modeChange = e => {
    e.preventDefault();
    const value = e.target.id;
    setAnchorEl(null);
    setError(null);
    if (mode === value) {
      setMode(null);
    } else {
      setMode(value);
    }
    dispatch(initializeForm('newPwForm'));
    editFormRefresh();
  };

  const pwChangeSubmit = e => {
    e.preventDefault();
    const { newPassword, newPasswordConfirm } = newPwForm;
    const parameter = {};
    // 비밀번호 확인하는 로직 필요

    if (newPassword !== newPasswordConfirm) {
      setError('새비밀번호 확인이 맞지 않습니다');
      return;
    }
    parameter.password = newPassword;
    parameter.id = userId;
    dispatch(myinfoedit({ parameter, token }));
  };

  const withdrawalSubmit = e => {
    e.preventDefault();
    const { password, text } = withDrawalForm;
    // 비밀번호 확인하는 로직 필요
    console.log(password);
    if (text === '탈퇴') {
      dispatch(accountdelete({ userId, token }));
      setError('탈퇴가 완료되었습니다.');
    } else {
      setError('탈퇴를 입력하세요');
    }
  };

  const menuClick = e => {
    if (anchorEl) {
      setAnchorEl(null);
    } else {
      setAnchorEl(e.currentTarget);
    }
  };

  const myinfoEditSubmit = e => {
    e.preventDefault();
    const parameter = {};
    const infoEditObjKeys = Object.keys(infoEditObj);
    for (let i = 0; i < infoEditObjKeys.length; i += 1) {
      const key = infoEditObjKeys[i];
      if (infoObj[key] !== infoEditObj[key]) {
        parameter[key] = infoEditObj[key];
      }
    }

    if (parameter && Object.keys(parameter).length === 0) {
      setError('수정사항이 없습니다');
      return;
    }

    parameter.id = userId;
    dispatch(myinfoedit({ parameter, token }));
  };

  return (
    <>
      <WithdrawalDialog
        mode={mode}
        withDrawalForm={withDrawalForm}
        error={error}
        modeChange={modeChange}
        formChange={formChange}
        withdrawalSubmit={withdrawalSubmit}
      />
      <PwChangeDialog
        mode={mode}
        error={error}
        newPwForm={newPwForm}
        modeChange={modeChange}
        pwChangeSubmit={pwChangeSubmit}
        formChange={formChange}
      />
      <ProfilePage
        infoObj={infoObj}
        infoEditObj={infoEditObj}
        anchorEl={anchorEl}
        mode={mode}
        error={error}
        handleChange={handleChange}
        formChange={formChange}
        typeChange={typeChange}
        informationOpenAgreeChange={informationOpenAgreeChange}
        menuClick={menuClick}
        modeChange={modeChange}
        myinfoEditSubmit={myinfoEditSubmit}
        editFormRefresh={editFormRefresh}
        loading={loading}
      />
    </>
  );
};
export default withRouter(ProfilePageContainer);
