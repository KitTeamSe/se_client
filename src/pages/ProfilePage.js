import React from 'react';
import ProfilePageContainer from '../containers/ProfilePage/ProfilePageContainer';

const ProfilePage = () => {
  return <ProfilePageContainer />;
};

export default ProfilePage;
