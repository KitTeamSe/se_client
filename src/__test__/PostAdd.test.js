/* eslint-disable */

import React from 'react';
import { mount } from 'enzyme';
import ProfilePage from '../components/ProfilePage/ProfilePage';

const infoObj = null;
const infoEditObj = null;
const anchorEl = null;
const mode = null;
const loading = null;

describe('프로필 화면 스냅샷', () => {
  it('renders post write', () => {
    const wrapper = mount(<ProfilePage infoObj={infoObj} infoEditObj={infoEditObj} anchorEl={infoEditObj} mode={infoEditObj} loading={infoEditObj}/>);
    expect(wrapper).toMatchSnapshot();
  });
});


