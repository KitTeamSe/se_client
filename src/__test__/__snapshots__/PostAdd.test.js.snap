// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostAdd/> matches snapshot 1`] = `
<PropfilePage
  anchorEl={null}
  infoEditObj={null}
  infoObj={null}
  loading={null}
  mode={null}
>
  <styled.div>
    <div
      className="sc-gsTEea igwFIj"
    >
      <Styled(Component)>
        <ForwardRef(CircularProgress)
          className="sc-pGacB kEMnlS"
        >
          <MuiCircularProgressRoot
            className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary sc-pGacB kEMnlS"
            ownerState={
              Object {
                "className": "sc-pGacB kEMnlS",
                "color": "primary",
                "disableShrink": false,
                "size": 40,
                "thickness": 3.6,
                "value": 0,
                "variant": "indeterminate",
              }
            }
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <span
              className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary sc-pGacB kEMnlS css-18lrjg1-MuiCircularProgress-root"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <MuiCircularProgressSvg
                className="MuiCircularProgress-svg"
                ownerState={
                  Object {
                    "className": "sc-pGacB kEMnlS",
                    "color": "primary",
                    "disableShrink": false,
                    "size": 40,
                    "thickness": 3.6,
                    "value": 0,
                    "variant": "indeterminate",
                  }
                }
                viewBox="22 22 44 44"
              >
                <svg
                  className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <MuiCircularProgressCircle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx={44}
                    cy={44}
                    fill="none"
                    ownerState={
                      Object {
                        "className": "sc-pGacB kEMnlS",
                        "color": "primary",
                        "disableShrink": false,
                        "size": 40,
                        "thickness": 3.6,
                        "value": 0,
                        "variant": "indeterminate",
                      }
                    }
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  >
                    <circle
                      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx={44}
                      cy={44}
                      fill="none"
                      r={20.2}
                      strokeWidth={3.6}
                      style={Object {}}
                    />
                  </MuiCircularProgressCircle>
                </svg>
              </MuiCircularProgressSvg>
            </span>
          </MuiCircularProgressRoot>
        </ForwardRef(CircularProgress)>
      </Styled(Component)>
    </div>
  </styled.div>
</PropfilePage>
`;

exports[`게시글 관련 테스트 renders post write 1`] = `
<PropfilePage
  anchorEl={null}
  infoEditObj={null}
  infoObj={null}
  loading={null}
  mode={null}
>
  <styled.div>
    <div
      className="sc-gsTEea igwFIj"
    >
      <Styled(Component)>
        <ForwardRef(CircularProgress)
          className="sc-pGacB kEMnlS"
        >
          <MuiCircularProgressRoot
            className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary sc-pGacB kEMnlS"
            ownerState={
              Object {
                "className": "sc-pGacB kEMnlS",
                "color": "primary",
                "disableShrink": false,
                "size": 40,
                "thickness": 3.6,
                "value": 0,
                "variant": "indeterminate",
              }
            }
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <span
              className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary sc-pGacB kEMnlS css-18lrjg1-MuiCircularProgress-root"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <MuiCircularProgressSvg
                className="MuiCircularProgress-svg"
                ownerState={
                  Object {
                    "className": "sc-pGacB kEMnlS",
                    "color": "primary",
                    "disableShrink": false,
                    "size": 40,
                    "thickness": 3.6,
                    "value": 0,
                    "variant": "indeterminate",
                  }
                }
                viewBox="22 22 44 44"
              >
                <svg
                  className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <MuiCircularProgressCircle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx={44}
                    cy={44}
                    fill="none"
                    ownerState={
                      Object {
                        "className": "sc-pGacB kEMnlS",
                        "color": "primary",
                        "disableShrink": false,
                        "size": 40,
                        "thickness": 3.6,
                        "value": 0,
                        "variant": "indeterminate",
                      }
                    }
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  >
                    <circle
                      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx={44}
                      cy={44}
                      fill="none"
                      r={20.2}
                      strokeWidth={3.6}
                      style={Object {}}
                    />
                  </MuiCircularProgressCircle>
                </svg>
              </MuiCircularProgressSvg>
            </span>
          </MuiCircularProgressRoot>
        </ForwardRef(CircularProgress)>
      </Styled(Component)>
    </div>
  </styled.div>
</PropfilePage>
`;

exports[`프로필 화면 스냅샷 renders post write 1`] = `
<PropfilePage
  anchorEl={null}
  infoEditObj={null}
  infoObj={null}
  loading={null}
  mode={null}
>
  <styled.div>
    <div
      className="sc-gsTEea igwFIj"
    >
      <Styled(Component)>
        <ForwardRef(CircularProgress)
          className="sc-pGacB kEMnlS"
        >
          <MuiCircularProgressRoot
            className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary sc-pGacB kEMnlS"
            ownerState={
              Object {
                "className": "sc-pGacB kEMnlS",
                "color": "primary",
                "disableShrink": false,
                "size": 40,
                "thickness": 3.6,
                "value": 0,
                "variant": "indeterminate",
              }
            }
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <span
              className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary sc-pGacB kEMnlS css-18lrjg1-MuiCircularProgress-root"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <MuiCircularProgressSvg
                className="MuiCircularProgress-svg"
                ownerState={
                  Object {
                    "className": "sc-pGacB kEMnlS",
                    "color": "primary",
                    "disableShrink": false,
                    "size": 40,
                    "thickness": 3.6,
                    "value": 0,
                    "variant": "indeterminate",
                  }
                }
                viewBox="22 22 44 44"
              >
                <svg
                  className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <MuiCircularProgressCircle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx={44}
                    cy={44}
                    fill="none"
                    ownerState={
                      Object {
                        "className": "sc-pGacB kEMnlS",
                        "color": "primary",
                        "disableShrink": false,
                        "size": 40,
                        "thickness": 3.6,
                        "value": 0,
                        "variant": "indeterminate",
                      }
                    }
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  >
                    <circle
                      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx={44}
                      cy={44}
                      fill="none"
                      r={20.2}
                      strokeWidth={3.6}
                      style={Object {}}
                    />
                  </MuiCircularProgressCircle>
                </svg>
              </MuiCircularProgressSvg>
            </span>
          </MuiCircularProgressRoot>
        </ForwardRef(CircularProgress)>
      </Styled(Component)>
    </div>
  </styled.div>
</PropfilePage>
`;
